datasets/tedlium2.py

```diff
@@ -0,0 +1,222 @@
+            ';; LABEL "male" "Male" "Male Talkers"',
+            ';; LABEL "female" "Female" "Female Talkers"',
+            ";;",
+        ]
+        for corpus_key in ["train", "dev", "test"]:
+            f = open("%s.stm" % corpus_key, "w")
+            f.write("\n".join(header) + "\n")
+
+            stm_folder = os.path.join(self.corpus_folders[corpus_key].get_path(), "stm")
+            for stm_file in sorted(os.listdir(stm_folder)):
+                if not stm_file.endswith(".stm"):
+                    continue
+                data = self.load_stm_data(os.path.join(stm_folder, stm_file))
+                for idx in range(len(data)):
+                    # some text normalization
+                    text = data[idx][6]
+                    text = text.replace("imiss", "i miss")
+                    text = text.replace("uptheir", "up their")
+                    text = re.sub(
+                        "(\w)'([a-zA-Z])", r"\1 '\2", text
+                    )  # split apostrophe
+                    data[idx][6] = text
+
+                    # train-only: segment boundary non-overlapping extension (kaldi)
+                    if corpus_key == "train":
+                        pre_seg = None if idx == 0 else data[idx - 1]
+                        next_seg = None if idx == len(data) - 1 else data[idx + 1]
+                        data[idx][3], data[idx][4] = extend_segment_time(
+                            data[idx], pre_seg, next_seg, 0.15, 0.1
+                        )
+
+                for seg in data:
+                    f.write(" ".join(seg) + "\n")
+
+            f.close()
+            shutil.move(
+                "%s.stm" % corpus_key, self.out_stm_files[corpus_key].get_path()
+            )
+
+    def make_corpus(self):
+        """
+        create bliss corpus from stm file (always include speakers)
+        """
+        for corpus_key in ["train", "dev", "test"]:
+            audio_dir = os.path.join(self.corpus_folders[corpus_key].get_path(), "sph")
+            stm_file = self.out_stm_files[corpus_key].get_path()
+            data = self.load_stm_data(stm_file)
+
+            c = corpus.Corpus()
+            c.name = "TED-LIUM-realease2"
+
+            speakers = []
+            last_rec_name = ""
+            recording = None
+            for seg in data:
+                rec_name, channel, spk_name, start, end, gender, text = seg
+
+                if not spk_name in speakers:
+                    speakers.append(spk_name)
+                    speaker = corpus.Speaker()
+                    speaker.name = spk_name
+                    if "female" in gender:
+                        speaker.attribs["gender"] = "female"
+                    elif "male" in gender:
+                        speaker.attribs["gender"] = "male"
+                    c.add_speaker(speaker)
+
+                if rec_name != last_rec_name:
+                    if recording:
+                        c.recordings.append(recording)
```

Comment:
```suggestion
                        c.add_recording(recording)
```
this function also registers `c` as `recording.corpus`

Reply:
this was also just copied from datasets/librispeech
changed now